app/src/shared/modules/integrationKey/services.js

```diff
@@ -18,8 +18,12 @@ export const getIntegrationKeys = (): ApiResult<Array<IntegrationKey>> => get({
     url: formatApiUrl('integration_keys'),
 })
 
-export const createPrivateKey = (name: string, privateKey: Address): ApiResult<IntegrationKey> =>
-    post({
+export const createPrivateKey = (name: string): ApiResult<IntegrationKey> => {
+    const web3 = getPublicWeb3()
```

Comment:
Public web3 instance is used, no need to open metamask.. let me know using the public instance is not good for this